src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -414,36 +360,26 @@ compatibility_subject_version(server::request_t rq, server::reply_t rp) {
 
     auto version = invalid_schema_version;
     if (ver == "latest") {
-        auto versions = rq.service().schema_store().get_versions(
-          req.sub, include_deleted::no);
-        if (versions.has_error()) {
-            rp.rep = make_errored_body(versions.error());
-            co_return rp;
-        }
-        if (versions.value().empty()) {
-            auto code = make_error_code(error_code::subject_version_not_found);
-            rp.rep = make_errored_body(code);
-            co_return rp;
+        auto versions = rq.service()
+                          .schema_store()
+                          .get_versions(req.sub, include_deleted::no)
+                          .value();
+        if (versions.empty()) {
+            throw as_exception(not_found(req.sub, version));
         }
-        version = versions.value().back();
+        version = versions.back();
     } else {
-        auto res = parse::from_chars<schema_version>{}(ver);
-        if (res.has_error()) {
-            rp.rep = make_errored_body(res.error());
-            co_return rp;
-        }
-        version = res.value();
+        version = parse::from_chars<schema_version>{}(ver).value();
     }
 
-    auto get_res = rq.service().schema_store().is_compatible(
-      req.sub, version, req.payload.schema, req.payload.type);
-    if (get_res.has_error()) {
-        rp.rep = make_errored_body(get_res.error());
-        co_return rp;
-    }
+    auto get_res = rq.service()
```

Comment:
this `linq` style api is beautiful! great work!

Reply:
I think that just the way clang format treats a train wreck.

Reply:
haha